### PR TITLE
feedback-only: Show automigration message on the Top Site, fix #2616

### LINKED ISSUE #2616: Show automigration (datamigration) description on the activity stream page
Interact with automigrate module to show the migration description on activity stream page

(Correspondent to [bug 1361287](https://bugzilla.mozilla.org/show_bug.cgi?id=1361287))

The flow (defined in [bug 1361286](https://bugzilla.mozilla.org/show_bug.cgi?id=1361286)) (need convert to the redux structure) includes
- Send message in `page.js` to remotepage `aboutNewTab.jsm`(`NewTab:MaybeShowUndoMessage`)
- `page.js` show the migrated description while receive `NewTab:Automigrated` message from `aboutNewTab.jsm`
  - trigger `keep` action (`NewTab:MigrationIsDone`) when user select `Keep It` button
    - hide all messages while receive `NewTab:HideMigrationMessage` message from `aboutNewTab.jsm`
 - show `manual import message` while receive `NewTab:MigrationIsReverted` message
  - trigger `undo` action (`NewTab:UndoAutoMigration`) when user select `undo` button
 - trigger `manual import` action ("NewTab:ManualMigration") when user select `manual import` link

Here's the workflow-correct screencast [bug 1361286](https://bugzilla.mozilla.org/show_bug.cgi?id=1361286)

### PR DESCRIPTION
**Pull request by gasolin**
Here's the feedback only PR which follows the current UX spec https://mozilla.invisionapp.com/share/Y3BGDY88H#/screens/230329192 

Implement based on #2616 comment https://github.com/mozilla/activity-stream/issues/2616#issuecomment-303605990

test instruction:
* build activity-stream and run in mozilla-central
* pref on activity-stream and `browser.migrate.automigrate.inpage.ui.enabled` while bug 1361286 is landed

Known Issues (Need feedback):
* only implement Main -> Content process now, because I can't show `this.props.Automigrate` state successfully in `Automigrate` component. It always returns `this.props.Automigrate is not found`

### CODE REVIEW

**Comment by gasolin** [comment]
@k88hudson could you kindly help me find why the state is not exist in the content page's store?

Any hint to debug the content page's store would also be helpful.

File: system-addon/content-src/components/AutoMigratePrompt/AutoMigratePrompt.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    //   }
+    // } else {
+    //   return;
+    // }
+  }
+}
+
+module.exports = connect(state => ({AutoMigrate: state.AutoMigrate}))(injectIntl(AutoMigratePrompt));
+module.exports._unconnected = AutoMigratePrompt;
+module.exports = AutoMigratePrompt;
```
Comment (Mardak) [inline system-addon/content-src/components/AutoMigratePrompt/AutoMigratePrompt.jsx]: This last line is overriding the previous two?

Comment (Mardak) [inline system-addon/content-src/components/AutoMigratePrompt/AutoMigratePrompt.jsx]: Removing that last line and adding a `console.log(this.props)` in `render`:

`Object { AutoMigrate: Object, dispatch: function messageMiddleware/</<(), intl: Object }`
where `"AutoMigrate":{"display":false,"stage":0,"msg":1}`

File: system-addon/lib/AutoMigrateFeed.jsm [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+Cu.import("resource://gre/modules/Log.jsm");
+let console = Log.repository.getLogger("AutoMigrate");
+console.addAppender(new Log.ConsoleAppender(new Log.BasicFormatter()));
+
+XPCOMUtils.defineLazyModuleGetter(this, "AutoMigrate",
+  "resource:///modules/AutoMigrate.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "Services",
+  "resource://gre/modules/Services.jsm");
+
+this.AutoMigrateFeed = class NewTabInit {
```
Comment (Mardak) [inline system-addon/lib/AutoMigrateFeed.jsm]: I believe the actually exported symbol for Firefox side uses the `class` name, so it shouldn't be `NewTabInit` here.

File: system-addon/lib/AutoMigrateFeed.jsm [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  onAction(action) {
+    switch (action.type) {
+      case at.NEW_TAB_VISIBLE:
+        AutoMigrate.shouldShowMigratePrompt(action._target.browser).then(prompt => {
```
Comment (Mardak) [inline system-addon/lib/AutoMigrateFeed.jsm]: Is this a new function to be implemented in nightly?

Comment (Mardak) [inline system-addon/lib/AutoMigrateFeed.jsm]: Oh yes indeed. It's in autoland already https://bugzilla.mozilla.org/show_bug.cgi?id=1361286

File: system-addon/content-src/components/AutoMigratePrompt/AutoMigratePrompt.jsx (hunk 1 above)
Comment (gasolin) [inline system-addon/content-src/components/AutoMigratePrompt/AutoMigratePrompt.jsx]: Thanks, it works fine after removed the last line! 🤦

**Pull request closed without merging**